.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Pull Request Prerequisite Checks
+
+on:
+  pull_request:
+    branches:
+      - main
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
Isn't this the default?

Reply:
@santam85  As this is triggered on pull request I am explicitly setting it to head reference of the PR to get till the last commit of the feature branch.
I am not sure what will it default to?

Reply:
actually it's not the default: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Basically the default is the merge commit of the PR, which is a better choice imho than the PR head (as that might not contain commits that are present on the target branch, failing to test the resulting merge and using the PR source branch instead)